presto-hive/src/main/java/com/facebook/presto/hive/HiveTableLayoutHandle.java

```diff
@@ -89,7 +89,7 @@ public HiveTableLayoutHandle(
         this(
                 schemaTableName,
                 tablePath,
-                partitionColumns,
+                partitionColumns.stream().map(BaseHiveColumnHandle.class::cast).collect(toList()),
```

Comment:
As discussed offline, we won't be able to make this change in the JSON constructor  since we don't want the information to get lost while serializing.